src/lib/EventPublisher.js

```diff
@@ -140,25 +151,30 @@ class EventPublisher extends EventEmitter {
           timeoutHandler();
         }
       }, subscriptionTimeout);
-
-      this.once(eventNamePrefix, (): void => clearTimeout(timeout));
+      this.once(eventNamePrefix, () => {
+        clearTimeout(timeout);
+      });
     }
 
     if (once) {
       this.once(eventNamePrefix, listener);
+      this.once(eventNamePrefix, () => {
+        this._subscriptionsByID.delete(subscriptionID);
+      });
     } else {
       this.on(eventNamePrefix, listener);
     }
-
     return subscriptionID;
   };
 
   unsubscribe = (subscriptionID: string) => {
-    const { eventNamePrefix, listener } = nullthrows(
-      this._subscriptionsByID.get(subscriptionID),
+    const subscription: Subscription = this._subscriptionsByID.get(
+      subscriptionID,
     );
-
-    this.removeListener(eventNamePrefix, listener);
+    if (!subscription) {
+      return;
```

Comment:
i cant interpret the error, i just took you suggestion to deal with already remove subscriptions.
maybe you can help me to understand what's wrong
```
const subscription: Subscription = this._subscriptionsByID.get(
170:       subscriptionID,
171:     );
         ^ undefined. This type is incompatible with
169:     const subscription: Subscription = this._subscriptionsByID.get(
                             ^^^^^^^^^^^^ object type
```